launch/launch/substitutions/command.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the Command substitution."""
+
+import shlex
+import subprocess
+from typing import Iterable
+from typing import List
+from typing import Text
+
+from .substitution_failure import SubstitutionFailure
+from ..frontend.expose import expose_substitution
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitution import Substitution
+
+
+@expose_substitution('command')
+class Command(Substitution):
+    """
+    Substitution that gets the output of a command as a string.
+
+    If the command is not found, fails, or has `stderr` output,
+    a `SubstitutionFailure` error is raised.
+    """
+
+    def __init__(
+        self,
+        command: SomeSubstitutionsType,
+    ) -> None:
+        """
+        Construct a command substitution.
+
+        :param command: command to be executed. The substitutions will be performed, and
+            `shlex.split` is applied to the result to form the arguments.
```

Comment:
@ivanpauno consider:

```
After substitutions are performed, `shlex.split` will be used to all command arguments.
```

Reply:
Yes, `shlex.split` is applied just once.
I updated to:
```
`shlex.split` will be used on the result.
```